### PR TITLE
[ETFE-3632] select-shipping-mark page

### PR DESCRIPTION
**Pull request by james-work-account**
Note: this PR doesn't implement navigation as that will be dealt with in a separate PR.

### CODE REVIEW

File: app/pages/sections/items/ItemsSection.scala [hunk 1]
```diff
@@ -38,4 +38,23 @@ case object ItemsSection extends Section[JsObject] {
...
+        itemsCount =>
+          (0 until itemsCount)
+            .flatMap(itemIdx => request.userAnswers.get(ItemsPackagingCount(itemIdx)).map {
+              packagingCount =>
+                (0 until packagingCount)
+                  .flatMap(packagingIdx => request.userAnswers.get(ItemPackagingShippingMarksPage(itemIdx, packagingIdx)))
+            })
+      }
+      .map(_.flatten.distinct)
+      .getOrElse(Seq())
```
Comment (MJCallahanPage) [inline app/pages/sections/items/ItemsSection.scala:59]: This has made me think about the following situations, what happens if:

- Item gets removed which was the one which initially had the shipping code?
- Item packaging gets removed which was the one which had the shipping code?
- Item packaging shipping code is then changed, after being selected in this drop down?

Would these scenarios invalidate the data if the user has entered if they picked one of the values which has either changed or no longer exists?

**Comment by james-work-account** [comment]
Got some outstanding queries so moving to Draft for now. UCD have confirmed that if PackagingB uses the value from PackagingA, then PackagingA's value changes, then PackagingB's value should also change. This throws up a few questions (such as the ones Matt raised above), which I'm chasing up.

File: app/pages/sections/items/ItemsSection.scala (hunk 1 above)
Comment (james-work-account) [reply app/pages/sections/items/ItemsSection.scala:59]: a. Not sure; asked UCD
b. Not sure; asked UCD
c. The change should be reflected in this item/packaging, so my current solution of copying the value into a new variable doesn't work

**Review by MJCallahanPage — APPROVED** [review]
LGTM 🟢

File: app/pages/sections/items/ItemsSection.scala (hunk 1 above)
Comment (james-work-account) [reply app/pages/sections/items/ItemsSection.scala]: a. Will be dealt with in an alignment
b. Will be dealt with in an alignment
c. Fixed

File: app/controllers/sections/items/ItemPackagingSelectShippingMarkController.scala [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+      form = form,
+      action = routes.ItemPackagingSelectShippingMarkController.onSubmit(request.ern, request.draftId, itemsIndex, packagingIdx, mode),
+      itemIdx = itemsIndex,
+      packagingIdx = packagingIdx,
+      selectOptions = selectItems
+    )))
+  }
+
+  private def withShippingMarks(f: Seq[String] => Future[Result])(implicit request: DataRequest[_]): Future[Result] = {
+    // TODO: handle what happens when Seq is empty. Currently the empty Seq is returned. Do we need to redirect somewhere?
```
[the remark this answers is not in the document]
Comment (james-work-account) [reply app/controllers/sections/items/ItemPackagingSelectShippingMarkController.scala]: Fixed in https://github.com/hmrc/emcs-tfe-create-movement-frontend/pull/272/commits/ba50c343c1e43b8543f279766ddc20c283f956cf

**Pull request merged**